workflow/controller/controller.go

```diff
@@ -482,28 +483,15 @@ func (wfc *WorkflowController) processNextItem() bool {
 
 	woc := newWorkflowOperationCtx(wf, wfc)
 
-	// Loading running workflow from persistence storage if nodeStatusOffload enabled
-	if wf.Status.IsOffloadNodeStatus() {
-		nodes, err := wfc.offloadNodeStatusRepo.Get(string(wf.UID), wf.GetOffloadNodeStatusVersion())
-		if err != nil {
-			woc.log.Errorf("getting offloaded nodes failed: %v", err)
-			woc.markWorkflowError(err, true)
-			woc.persistUpdates()
-			wfc.throttler.Remove(key)
-			return true
-		}
-		woc.wf.Status.Nodes = nodes
-	}
-
-	// Decompress the node if it is compressed
-	err = packer.DecompressWorkflow(woc.wf)
+	err = wfc.hydrator.Hydrate(woc.wf)
 	if err != nil {
-		woc.log.Errorf("workflow decompression failed: %v", err)
+		woc.log.Errorf("hydration failed: %v", err)
 		woc.markWorkflowError(err, true)
```

Comment:
markWorkflowError tries to access the workflow nodes, which may not exist if the hydration has failed.

Reply:
catch-22: if we cannot hydrate the workflow, then we cannot access the nodes